sync/awsdatabricks.py

```diff
@@ -388,10 +382,67 @@ def write_file(body: bytes):
                     "Reservations": [*removed_instances, *updated_instances, *new_instances]
                 }
 
-            write_file(orjson.dumps(instances))
+            # Now record the ebs volumes
+            volumes = _get_ebs_volumes(cluster_id, ec2)
+
+            for v in volumes:
+                if v["VolumeId"] not in recorded_volume_id_set:
+                    recorded_volume_id_set.add(v["VolumeId"])
+                    recorded_volumes.append(v)
 
+            instances["Volumes"] = recorded_volumes
+
+            write_file(orjson.dumps(instances))
             previous_instances = instances
         except Exception as e:
             logger.error(f"Exception encountered while polling cluster: {e}")
 
         sleep(polling_period)
+
+
+def _get_ebs_volumes(cluster_id: str, ec2_client: "botocore.client.ec2") -> List[Dict]:
+
+    args = {}
+    volumes = []
+    while True:
```

Comment:
For both this function and `_get_ec2_instances`, I personally prefer a pattern more like -

```python
response = ec2_client.describe_volumes(...)
volumes = response.get("Volumes", [])
next_token = response.get("NextToken")

while next_token:
    response = ec2_client.describe_volumes(..., NextToken=next_token)
    volumes += response.get("Volumes", [])
    next_token = response.get("NextToken")

return volumes
```

This eliminates the need for a `break` in your loop, and explicitly separates the initial load logic from the paging logic

Reply:
I will add that I used to always try to put the initial / paging logic in the same loop, but I've given up on that. Separating out the two now generally, I find, leads to code that is easier to follow / update

Reply:
Happy to make this change. Just curious, does loop breaking like this make you nervous from a brittleness standpoint?